src/filtertransdialog.cpp

```diff
@@ -1244,9 +1246,19 @@ bool mmFilterTransactionsDialog::mmIsRecordMatches(const Model_Checking::Data &t
     else if (mmIsNumberChecked() && (mmGetNumber().empty() ? !tran.TRANSACTIONNUMBER.empty()
         : tran.TRANSACTIONNUMBER.empty() || !tran.TRANSACTIONNUMBER.Lower().Matches(mmGetNumber().Lower())))
         ok = false;
-    else if (mmIsNotesChecked() && (mmGetNotes().empty() ? !tran.NOTES.empty()
-        : tran.NOTES.empty() || !tran.NOTES.Lower().Matches(mmGetNotes().Lower())))
-        ok = false;
+    else if (mmIsNotesChecked()) {
+        if (mmGetNotes().empty() && !tran.NOTES.empty()) ok = false;
+        else if (!mmGetNotes().empty()) {
+            if (tran.NOTES.empty()) ok = false;
+            else if (mmGetNotes().StartsWith("regex[")) {
+                std::string pattern = mmGetNotes().SubString(6, mmGetNotes().find_last_of(']') - 1).ToStdString();
+                std::regex regex(pattern, std::regex_constants::icase);
+                if (!std::regex_search(tran.NOTES.ToStdString(), regex)) ok = false;
+                notesEdit_->SetValue("regex[" + pattern + "]");
```

Comment:
What's the reasoning behind this change to the field?

Reply:
The pattern is derived from whatever is between the [ ], but the user could type something like "regex[...]extratexthere". In this code the pattern is still the [...] and the "extratexthere" is simply ignored, but when you see the transaction report under "Filtering Details" it shows the filter is "Notes: regex[...]extratexthere" rather than "Notes: regex[...]", so this bit was to get the report display in sync with the applied filter.

Reply:
Switching to regex: would eliminate the need for this since we don't need to find the end of the pattern string.